File: apps/pos/src/lib/order-type-label.test.ts
import { describe, it, expect } from 'vitest';
import { formatOrderTypeLabel } from './order-type-label';

describe('formatOrderTypeLabel', () => {
  it('dine_in without partner shows "Dine-in"', () => {
    expect(
      formatOrderTypeLabel({
        type: 'dine_in',
        deliveryPartnerTitle: null,
        deliveryExternalRef: null,
      }),
    ).toBe('Dine-in');
  });

  it('dine_in ignores partner fields', () => {
    expect(
      formatOrderTypeLabel({
        type: 'dine_in',
        deliveryPartnerTitle: 'HungerStation',
        deliveryExternalRef: 'HS-1',
      }),
    ).toBe('Dine-in');
  });

  it('takeaway without partner shows "Takeaway"', () => {
    expect(
      formatOrderTypeLabel({
        type: 'takeaway',
        deliveryPartnerTitle: null,
        deliveryExternalRef: null,
      }),
    ).toBe('Takeaway');
  });

  it('takeaway with empty title shows "Takeaway"', () => {
    expect(
      formatOrderTypeLabel({
        type: 'takeaway',
        deliveryPartnerTitle: '',
        deliveryExternalRef: null,
      }),
    ).toBe('Takeaway');
  });

  it('takeaway with partner and no ref shows the title', () => {
    expect(
      formatOrderTypeLabel({
        type: 'takeaway',
        deliveryPartnerTitle: 'HungerStation',
        deliveryExternalRef: null,
      }),
    ).toBe('HungerStation');
  });

  it('takeaway with partner and empty ref shows the title only', () => {
    expect(
      formatOrderTypeLabel({
        type: 'takeaway',
        deliveryPartnerTitle: 'HungerStation',
        deliveryExternalRef: '',
      }),
    ).toBe('HungerStation');
  });

  it('takeaway with partner and ref shows "title / ref"', () => {
    expect(
      formatOrderTypeLabel({
        type: 'takeaway',
        deliveryPartnerTitle: 'HungerStation',
        deliveryExternalRef: 'HS-883129',
      }),
    ).toBe('HungerStation / HS-883129');
  });

  it('trims whitespace around the external ref', () => {
    expect(
      formatOrderTypeLabel({
        type: 'takeaway',
        deliveryPartnerTitle: 'HungerStation',
        deliveryExternalRef: '  HS-1  ',
      }),
    ).toBe('HungerStation / HS-1');
  });

  it('missing partner fields are treated as no partner', () => {
    expect(formatOrderTypeLabel({ type: 'takeaway' })).toBe('Takeaway');
  });

  it('whitespace-only title is treated as no partner', () => {
    expect(
      formatOrderTypeLabel({
        type: 'takeaway',
        deliveryPartnerTitle: '   ',
        deliveryExternalRef: 'HS-1',
      }),
    ).toBe('Takeaway');
  });
});

File: apps/android/app/src/test/java/com/spicyhome/pos/util/OrderTypeLabelTest.kt
package com.spicyhome.pos.util

import com.google.common.truth.Truth.assertThat
import org.junit.Test

/**
 * Kotlin twin of the POS order-type label tests — mirrors the same acceptance
 * matrix (see `apps/pos/src/lib/order-type-label.test.ts`) so the two
 * implementations cannot silently diverge.
 */
class OrderTypeLabelTest {

    // ── Acceptance matrix (mirrors the POS example table) ───────────────

    @Test
    fun `dine_in without partner shows Dine-in`() {
        assertThat(OrderTypeLabel.format("dine_in")).isEqualTo("Dine-in")
    }

    @Test
    fun `dine_in ignores partner fields`() {
        assertThat(OrderTypeLabel.format("dine_in", "HungerStation", "HS-1"))
            .isEqualTo("Dine-in")
    }

    @Test
    fun `takeaway without partner shows Takeaway`() {
        assertThat(OrderTypeLabel.format("takeaway")).isEqualTo("Takeaway")
    }

    @Test
    fun `takeaway with empty title shows Takeaway`() {
        assertThat(OrderTypeLabel.format("takeaway", "", null)).isEqualTo("Takeaway")
    }

    @Test
    fun `takeaway with title and no ref shows title only`() {
        assertThat(OrderTypeLabel.format("takeaway", "HungerStation", null))
            .isEqualTo("HungerStation")
    }

    @Test
    fun `takeaway with title and empty ref shows title only`() {
        assertThat(OrderTypeLabel.format("takeaway", "HungerStation", ""))
            .isEqualTo("HungerStation")
    }

    @Test
    fun `takeaway with title and ref shows title slash ref`() {
        assertThat(OrderTypeLabel.format("takeaway", "HungerStation", "HS-883129"))
            .isEqualTo("HungerStation / HS-883129")
    }

    @Test
    fun `takeaway trims whitespace around ref`() {
        assertThat(OrderTypeLabel.format("takeaway", "HungerStation", "  HS-1  "))
            .isEqualTo("HungerStation / HS-1")
    }

    // ── Extra edge cases ────────────────────────────────────────────────

    @Test
    fun `takeaway with whitespace-only title shows Takeaway`() {
        assertThat(OrderTypeLabel.format("takeaway", "   ", "HS-1"))
            .isEqualTo("Takeaway")
    }

    @Test
    fun `takeaway with null title and ref shows Takeaway`() {
        assertThat(OrderTypeLabel.format("takeaway", null, null))
            .isEqualTo("Takeaway")
    }

    @Test
    fun `takeaway with title null and ref set shows Takeaway (no partner)`() {
        assertThat(OrderTypeLabel.format("takeaway", null, "HS-1"))
            .isEqualTo("Takeaway")
    }

    @Test
    fun `takeaway trims whitespace around title`() {
        assertThat(OrderTypeLabel.format("takeaway", "  HungerStation  ", null))
            .isEqualTo("HungerStation")
    }

    @Test
    fun `unknown non-takeaway type shows Dine-in`() {
        assertThat(OrderTypeLabel.format("delivery", "HungerStation", "HS-1"))
            .isEqualTo("Dine-in")
    }

    @Test
    fun `takeaway with whitespace-only ref shows title only`() {
        assertThat(OrderTypeLabel.format("takeaway", "HungerStation", "   "))
            .isEqualTo("HungerStation")
    }
}
